src/cpu/smp.rs

```diff
@@ -243,14 +243,12 @@ pub fn smp_run_bios_vmpl() -> bool {
             return false;
         }
 
-        let vmsa_va: VirtAddr = match pgtable_map_pages_private(vmsa_pa, PAGE_SIZE) {
+        let vmsa_map: MapGuard = match MapGuard::new_private(vmsa_pa, PAGE_SIZE, true) {
             Ok(r) => r,
             Err(_e) => return false,
         };
 
-        vc_ap_create(vmsa_va, PERCPU.apic_id());
-
-        pgtable_unmap_pages(vmsa_va, PAGE_SIZE);
+        vc_ap_create(vmsa_map.va, PERCPU.apic_id());
```

Comment:
Use `funcs!` in the implementation of MapGuard to retrieve the VA using the getter (`vmsa_map.va()`). Fields `va` and `len` will not need to be pub anymore.

Reply:
You're right, I'll change that.

BTW - for another issue - [Rust naming conventions](https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter) say that getters should not begin with `get_`. Not sure if we want to adhere to that.